File: Reverse_Prefix/reverse.go
package reverseprefix

func reversePrefix(word string, ch byte) string {
	byteArr := []byte(word)
	isFound, index := false, 0
	for ; index < len(byteArr); index++ {
		if byteArr[index] == ch {
			isFound = true
			break
		}
	}

	if !isFound {
		return word
	}

	for i, j := 0, index; i < j; i, j = i+1, j-1 {
		byteArr[i], byteArr[j] = byteArr[j], byteArr[i]
	}

	return string(byteArr)
}

File: Reverse_Prefix/reverse.py
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        is_found, index = False, 0

        while index < len(word):
            if word[index] == ch:
                is_found = True
                break
            index += 1

        if not is_found:
            return word

        char_list = list(word)
        i, j = 0, index

        while i < j:
            char_list[i], char_list[j] = char_list[j], char_list[i]

            i, j = i + 1, j - 1
        return "".join(char_list)
